politeiawww/dcc.go

```diff
@@ -899,3 +910,139 @@ func (p *politeiawww) getDCCComments(token string) ([]www.Comment, error) {
 
 	return comments, nil
 }
+
+func (p *politeiawww) processSetDCCStatus(ad cms.SetDCCStatus, u *user.User) (*cms.SetDCCStatusReply, error) {
+	log.Tracef("processSetDCCStatus: %v", u.PublicKey())
+
+	// Validate signature
+	msg := fmt.Sprintf("%v%v%v", ad.Token, int(ad.Status), ad.Reason)
+	err := validateSignature(ad.PublicKey, ad.Signature, msg)
+	if err != nil {
+		return nil, err
+	}
+
+	dcc, err := p.getDCC(ad.Token)
+	if err != nil {
+		if err == cache.ErrRecordNotFound {
+			err = www.UserError{
+				ErrorCode: cms.ErrorStatusDCCNotFound,
+			}
+		}
+		return nil, err
+	}
+
+	err = validateDCCStatusTransition(dcc.Status, ad.Status)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the change record.
+	c := backendDCCStatusChange{
+		Version:        backendDCCStatusChangeVersion,
+		AdminPublicKey: u.PublicKey(),
+		Timestamp:      time.Now().Unix(),
+		NewStatus:      ad.Status,
+		Reason:         ad.Reason,
+		Signature:      ad.Signature,
+	}
+	blob, err := encodeBackendDCCStatusChange(c)
+	if err != nil {
+		return nil, err
+	}
+
+	challenge, err := util.Random(pd.ChallengeSize)
+	if err != nil {
+		return nil, err
+	}
+
+	pdCommand := pd.UpdateVettedMetadata{
+		Challenge: hex.EncodeToString(challenge),
+		Token:     ad.Token,
+		MDAppend: []pd.MetadataStream{
+			{
+				ID:      mdStreamDCCStatusChanges,
+				Payload: string(blob),
+			},
+		},
+	}
+
+	responseBody, err := p.makeRequest(http.MethodPost, pd.UpdateVettedMetadataRoute, pdCommand)
+	if err != nil {
+		return nil, err
+	}
+
+	var pdReply pd.UpdateVettedMetadataReply
+	err = json.Unmarshal(responseBody, &pdReply)
+	if err != nil {
+		return nil, fmt.Errorf("Could not unmarshal UpdateVettedMetadataReply: %v",
+			err)
+	}
+
+	// Verify the UpdateVettedMetadata challenge.
+	err = util.VerifyChallenge(p.cfg.Identity, challenge, pdReply.Response)
+	if err != nil {
+		return nil, err
+	}
+
+	dbDCC, err := p.cmsDB.DCCByToken(ad.Token)
+	if err != nil {
+		return nil, err
+	}
+	dbDCC.Status = ad.Status
+	dbDCC.StatusChangeReason = ad.Reason
+
+	// Update cmsdb
+	err = p.cmsDB.UpdateDCC(dbDCC)
+	if err != nil {
+		return nil, err
+	}
+
+	// Only do further processing if it was an approved DCC
+	if ad.Status == cms.DCCStatusApproved {
+		if dcc.DCC.Type == cms.DCCTypeIssuance {
+			// Do DCC user Issuance processing
+			err := p.issuanceDCCUser(dcc.DCC.NomineeUserID, u.ID.String(), int(dcc.DCC.Domain), int(dcc.DCC.ContractorType))
+			if err != nil {
+				return nil, err
+			}
+			return &cms.SetDCCStatusReply{}, nil
```

Comment:
```suggestion
```

Don't think this is actually doing anything here since its going to return a on line 1019 anyway.